core/payment/src/dao/debit_note.rs

```diff
@@ -236,22 +237,28 @@ impl<'c> DebitNoteDao<'c> {
                 .find((&debit_note_id, &owner_id))
                 .select((dsl::activity_id, dsl::total_amount_due, dsl::role))
                 .first(conn)?;
-            update_status(
-                &vec![debit_note_id.clone()],
-                &owner_id,
-                &DocumentStatus::Accepted,
-                conn,
-            )?;
+            let mut events = vec![DebitNoteEventType::DebitNoteAcceptedEvent];
+            let status = if amount.0 == BigDecimal::from(0) {
```

Comment:
This line could use some comment. It looks a little bit hacky and someone reading this code might get confused.